core/src/test/java/dev/streamx/cli/ingestion/unpublish/UnpublishCommandTest.java

```diff
@@ -36,31 +43,74 @@ void setup() {
     initializeWiremock();
   }
 
-
-  @Test
-  public void shouldUnpublishUsingIngestionClient(QuarkusMainLauncher launcher) {
-    // when
-    LaunchResult result = launcher.launch("unpublish",
-        "--ingestion-url=" + getIngestionUrl(),
-        CHANNEL, KEY);
-
-    // then
-    assertThat(result.exitCode()).isZero();
+  @Nested
+  @QuarkusMainTest
+  @TestProfile(UnauthorizedProfile.class)
+  class UnauthorizedTest {
+
+    @Test
+    public void shouldUnpublishUsingIngestionClient(QuarkusMainLauncher launcher) {
+      // when
+      LaunchResult result = launcher.launch("unpublish",
+          "--ingestion-url=" + getIngestionUrl(),
+          CHANNEL, KEY);
+
+      // then
+      assertThat(result.exitCode()).isZero();
+    }
+
+    @Test
+    public void shouldUnpublishUsingUnauthorizedIngestionClient(QuarkusMainLauncher launcher) {
```

Comment:
same test for publish?

Reply:
Basically it would be nice to refactor and extract common ingestion behaviours, however I would like to do it in separated PR

Reply:
here you are
https://github.com/streamx-dev/streamx-cli/pull/50/files#diff-9687c80c617604e3e864746e212f14cbed2bbdf5ceb45ca7ba371121881b2f72R102

Reply:
the test is there, I didn't notice